src/app/flightsearch/flightsearch.component.ts

```diff
@@ -0,0 +1,538 @@
+    tempObject["destination"] = element[0].destination;
+    tempObject["fromDateTime"] = element[0].approximateDateTimeStart;
+    tempObject["toDateTime"] = element[0].approximateDateTimeEnd;
+    let duration = new Date(element[0].approximateDateTimeEnd).getTime() - new Date(element[0].approximateDateTimeStart).getTime();
+    let minutes = ((duration / (1000* 60)) % 60);
+    let hours = Math.floor(((duration / (1000 * 60 * 60)) % 24));
+    tempObject["duration"] = hours + "h " + minutes + "m";
+    tempObject["loyaltyPoints"] = element[0].possibleLoyaltyPoints;
+    tempObject["basePrice"] = formatter.format(element[0].seats[0].price);
+    tempObject["multihop"] = false;
+    tempObject["airplane"] = element[0].airplane;
+    tempObject["seats"] = element[0].seats;
+    let economySeats = element[0].seats.filter((x:any) => x.seatClass == "ECONOMY" && x.seatStatus=="AVAILABLE");
+    let businessSeats = element[0].seats.filter((x:any) => x.seatClass == "BUSINESS" && x.seatStatus=="AVAILABLE")
+    tempObject["economy"] = economySeats.length;
+    tempObject["business"] = businessSeats.length;
+    tempObject["creationTime"] = element[0].creationDateTime;
+    tempObject["numberOfHops"] = element.length;
+    tempObject["durationInMilliseconds"] = duration;
+    return tempObject;
+  }
+
+  initializePeopleString(): void{
+  this.numberOfPeople = this.adult + " Adult";
+  if (this.countOfChildren > 0){
+    this.numberOfPeople += "; " + this.countOfChildren + " Children"; 
+    }
+  }
+
+  toggleTicket(idNumber:any, event:any){
+
+    event.stopPropagation();
+
+    //Close all overlays
+    let overlays = document.getElementsByClassName("overlayForAdditionalDetails") as HTMLCollectionOf<HTMLElement>;
+    for (let i = 0; i < overlays.length; i++){
+      overlays[i].style.display = "none";
+    }
+
+    let idString = idNumber + "";
+    (document.getElementById(idString) as HTMLElement).style.display = "initial";
+  }
+
+  close(idNumber:any){
+    let idString = idNumber + "";
+    (document.getElementById(idString) as HTMLElement).style.display = "none";
+  }
+
+  eventSort(sortMethod:string, id:string){
+
+    let specificElement = document.getElementById(id) as HTMLElement;
+
+    //Checks if user want to remove the sort return back to default
+    if (specificElement.classList.contains("selected")){
+      specificElement.classList.remove("selected");
+      this.sort = "expensive";
+      this.sortData();
+      return;
+    }
+
+    //Else do these things
+    let elements = document.getElementsByClassName("hoverListItem") as HTMLCollectionOf<HTMLElement>;
+    for (let i = 0; i < elements.length; i++){
+      elements[i].classList.remove("selected");
+    }
+    
+    specificElement.classList.add("selected");
+    this.sort = sortMethod;
+    this.sortData();
+  }
+
+  sortData(){
+    let data = this.viewData;
+    switch(this.sort){
+      case "expensive":
```

Comment:
definitely externalize these magic strings into a constant value. Perfect use case for a [TypeScript Enum](https://www.typescriptlang.org/docs/handbook/enums.html)